tachyon/zk/base/entities/entity.h

```diff
@@ -49,6 +49,12 @@ class Entity {
   }
   crypto::Transcript<Commitment>* transcript() { return transcript_.get(); }
 
+  PCSTy&& TakePCS() { return std::move(pcs_); }
+  std::unique_ptr<Domain> TakeDomain() { return std::move(domain_); }
+  std::unique_ptr<ExtendedDomain> TakeExtendedDomain() {
+    return std::move(extended_domain_);
+  }
```

Comment:
It's just a question. Is it good to do as follows or not? I'm fine if you answer no, because `TakeXXX()` name implies some of members are going to be moved.

```suggestion
  PCSTy&& TakePCS() && { return std::move(pcs_); }
  std::unique_ptr<Domain> TakeDomain() && { return std::move(domain_); }
  std::unique_ptr<ExtendedDomain> TakeExtendedDomain() && {
    return std::move(extended_domain_);
  }
```

Reply:
I think it's good. But as we discussed, since we're going to have `ToVerifier` method in the future, I'll leave it as now.